lib/releases.sh

```diff
@@ -2,29 +2,49 @@
 
 function get_latest_release() {
   set +x
+  # get last page
   if [ -z "${GITHUB_TOKEN:-}" ]; then
-    release="$(curl -sL "${1}")" || ( set -x && exit 1 )
+    last_page=$(curl -s -I "${1}" -H "Authorization: token $GITHUB_TOKEN" | grep '^link:' | sed -e 's/^link:.*page=//g' -e 's/>.*$//g') || (set -x && exit 1)
   else
-    release="$(curl -H "Authorization: token ${GITHUB_TOKEN}" -sL "${1}")" || ( set -x && exit 1 )
+    last_page=$(curl -s -I "${1}" | grep '^link:' | sed -e 's/^link:.*page=//g' -e 's/>.*$//g') || (set -x && exit 1)
   fi
-  # This gets the latest release as vx.y.z or vx.y.z-rc.0, including any version with a suffix starting with - , for example -rc.0
-  # The order is exactly as released in Github.
-  # Downside is that selecting official releases only isn't possible, while pre-release
-  # selection is possible given specific enough prefix, like v1.3.0-pre
-  release_tag="$(echo "$release" | jq -r "[.[].tag_name | select( startswith(\"${2:-}\"))] | .[0]")"
+  # default last page to 1
+  last_page=${last_page:-1}
 
+  for current_page in $(seq 1 "$last_page"); do
```

Comment:
Any time you want to use `seq` to create sequences in Bash for loops, use `{1..$last_page}` instead. This avoids various issues with subshells, output etc.

Reply:
done

Reply:
(warning) Bash doesn't support variables in brace range expansions !

Reply:
Duh, my bad. See, Bash is hard :)